eng/tools/TypeSpecValidation/test/emit-autorest.ts

```diff
@@ -0,0 +1,61 @@
+import { join } from "path";
+import { EmitAutorestRule } from "../src/rules/emit-autorest.js";
+import { TsvTestHost } from "./tsv-test-host.js";
+import { strict as assert } from "node:assert";
+
+describe("emit-autorest", function () {
+  it("should succeed if no main.tsp", async function () {
+    let host = new TsvTestHost();
+    host.checkFileExists = async (file: string) => file != join(TsvTestHost.folder, "main.tsp");
+
+    const result = await new EmitAutorestRule().execute(host, TsvTestHost.folder);
+
+    assert(result.success);
+  });
+
+  it("should succeed if emits autorest", async function () {
+    let host = new TsvTestHost();
+    host.readTspConfig = async (_folder: string) => `
+emit:
+  - "@azure-tools/typespec-autorest"
+`;
+
+    const result = await new EmitAutorestRule().execute(host, TsvTestHost.folder);
+
+    assert(result.success);
+  });
+
+  it("should fail if no config", async function () {
+    let host = new TsvTestHost();
+    host.readTspConfig = async (_folder: string) => "";
```

Comment:
just want to double check that readfile will return empty string if no file is found

Reply:
It will throw, but I think it's fine.  Rule `FolderStructure` should already ensure `tspconfig.yaml` exists.